runners/apex/src/main/java/org/apache/beam/runners/apex/translators/functions/ApexParDoOperator.java

```diff
@@ -104,14 +102,16 @@ public ApexParDoOperator(
       List<TupleTag<?>> sideOutputTags,
       WindowingStrategy<?, ?> windowingStrategy,
       List<PCollectionView<?>> sideInputs,
-      Coder<WindowedValue<InputT>> inputCoder
+      Coder<WindowedValue<InputT>> inputCoder,
+      StateInternalsFactory<Void> stateInternalsFactory
```

Comment:
I've lost track for a moment - is this called before execution, or when execution begins?